docs/conf.py

```diff
@@ -108,15 +108,10 @@
 # -- General configuration ----------------------------------------------------
 
 # If your documentation needs a minimal Sphinx version, state it here.
-needs_sphinx = "3.0"
-
-# The intersphinx_mapping in sphinx_astropy.sphinx refers to astropy for
-# the benefit of other packages who want to refer to objects in the
-# astropy core.  However, we don't want to cyclically reference astropy in its
-# own build so we remove it here.
-del intersphinx_mapping["astropy"]
+needs_sphinx = "8.2.0"  # keep in sync with pyproject.toml
 
 # add any custom intersphinx for astropy
+intersphinx_resolve_self = "astropy"
```

Comment:
Yay! I'm very glad to see that this is in Sphinx. We raised this issue with them a while ago after we'd hacked this together.